File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

// Function for all change
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask for how much change is owned user input
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0); // Cant go into negtive

    int quarters = calculate_quarters(cents);

    cents = cents - (quarters * 25);

    int dimes = calculate_dimes(cents);

    cents = cents - (dimes * 10);

    int nickels = calculate_nickels(cents);

    cents = cents - (nickels * 5);

    int pennies = calculate_pennies(cents);

    int total_coins = quarters + dimes + nickels + pennies;
    printf("%d\n", total_coins);
}

int calculate_quarters(int cents)
{

    return cents / 25;
}

int calculate_dimes(int cents)
{

    return cents / 10;
}

int calculate_nickels(int cents)
{

    return cents / 5;
}

int calculate_pennies(int cents)
{

    return cents;
}

File: cash.py/cash.py
from cs50 import get_float

def main():
    # Loop until a valid input is received
    while True:
        # Get the change owed from the user
        change = get_float("Change owed: ")

        # If the input is non-negative, break the loop
        if change >= 0:
            break

    # Convert the change to cents (avoid floating point issues)
    cents = round(change * 100)

    # Initialize the coin count
    coins = 0

    # Calculate the number of coins needed (starting with the largest coin first)
    coins += cents // 25  # Quarters
    cents %= 25

    coins += cents // 10  # Dimes
    cents %= 10

    coins += cents // 5   # Nickels
    cents %= 5

    coins += cents // 1   # Pennies

    # Output the result (minimum number of coins)
    print(coins)

if __name__ == "__main__":
    main()
